python/CMakeLists.txt

```diff
@@ -1,72 +1,159 @@
-# Note: using same pattern used in cvc5 for scikit-build / python module checking
-# Check if given Python module is installed and raises a FATAL_ERROR error
-# if the module cannot be found.
+# First, ensure we have Python 3.9 or newer with all required components
+# Check for a virtualenv python first
+find_python_in_virtualenv()
+# Want python for building a module, so use Development.Module. See:
+# https://stackoverflow.com/questions/78495429/attempting-to-build-python-binary-modules-on-manylinux-find-packagepython3-on
+find_package(Python 3.9 REQUIRED COMPONENTS Interpreter Development.Module)
+
+# Helper function to check for required Python packages
 function(check_python_module module)
-  execute_process(
-    COMMAND
-    ${PYTHON_EXECUTABLE} -c "import ${module}"
-    RESULT_VARIABLE
-      RET_MODULE_TEST
-    ERROR_QUIET
-  )
-  set(module_name ${ARGN})
-  if(NOT module_name)
-    set(module_name ${module})
-  endif()
-
-  if(RET_MODULE_TEST)
-    message(FATAL_ERROR
-        "Could not find module ${module_name} for Python "
-        "version ${PYTHON_VERSION_MAJOR}.${PYTHON_VERSION_MINOR}. "
-        "Make sure to install ${module_name} for this Python version "
-        "via \n`${PYTHON_EXECUTABLE} -m pip install ${module_name}'.\n"
-        "Note: You need to have pip installed for this Python version.")
-  endif()
+    execute_process(
```

Comment:
GH shows something strange about these whitespaces. Maybe there are tabs?

Reply:
Hmm I see what you mean but I checked in my editor and it's definitely spaces. Maybe it's the other way? Might have been tabs before and my editor converted it to spaces.